core/src/main/java/org/jdbi/v3/core/statement/SqlStatement.java

```diff
@@ -1284,15 +1311,17 @@ public final This bindList(String key, Object... values) {
      *     .list();
      * </pre>
      *
-     * @param key    attribute name
-     * @param values list of values that will be comma-spliced into the defined attribute value.
+     * @param key     attribute name
+     * @param onEmpty handler for null/empty list
+     * @param values  list of values that will be comma-spliced into the defined attribute value.
      * @return this
      * @throws IllegalArgumentException if the list is empty.
+     * @see EmptyHandling
      */
-    public final This bindList(String key, List<?> values) {
-        if (values.isEmpty()) {
-            throw new IllegalArgumentException(
-                    getClass().getSimpleName() + ".bindList was called with an empty list.");
+    public final This bindList(String key, BiConsumer<SqlStatement, String> onEmpty, List<?> values) {
```

Comment:
should I actually maybe declare this as `Iterable` and add an overload for `Iterator` so that it aligns with `@BindList`?

Reply:
also, parameter order is not correct yet